core/vm/custom_eip.go

```diff
@@ -0,0 +1,152 @@
+package vm
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+)
+
+// OpCodeInfo contains information required to identify an EVM operation.
+type OpCodeInfo struct {
+	Number OpCode
+	Name   string
+}
+
+// Operation is an utility struct that wraps the private type
+// operation.
+type Operation struct {
+	Op *operation
+}
+
+// ExtendActivators allows to merge the go ethereum activators map
+// with additional custom activators.
+func ExtendActivators(eips map[int]func(*JumpTable)) error {
+	// Catch early duplicated eip.
+	keys := make([]int, 0, len(eips))
+	for k := range eips {
+		if ValidEip(k) {
+			return fmt.Errorf("duplicate activation: %d is already present in %s", k, ActivateableEips())
+		}
+		keys = append(keys, k)
+	}
+
+	// Sorting keys to ensure deterministic execution.
+	sort.Ints(keys)
+
+	for _, k := range keys {
+		activators[k] = eips[k]
+	}
+	return nil
+}
+
+// GetActivatorsEipNumbers returns the keys of EIPs registered in
+// the activators map.
+// Used only in tests.
+func GetActivatorsEipNumbers() []int {
+	keys := make([]int, 0, len(activators))
+	for k, _ := range activators {
+		keys = append(keys, k)
+	}
+
+	sort.Ints(keys)
+	return keys
+}
+
+// ExtendOperations returns an instance of the new operation and register it in the list
+// of available ones.
+// Return an error if an operation with the same name is already present.
+// This function is used to prevent the overwrite of an already existent operation.
+func ExtendOperations(
```

Comment:
Is this actually used anywhere apart from tests? Can't find another reference in the unforked `cosmos/evm` branch or in this repo apart from within `custom_eip_test.go`

Reply:
Yeah, seems like it's only used in tests? Not sure why that is. Fine with keeping it for now and evaluating whether to remove it when we try to simplify our patch set in the future.